### PR TITLE
feat(dlp): added a sample for dlp deidentify deterministic

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description

- Added a sample for [dlp deidentify deterministic](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-deterministic) with a unit test case.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid/deid_deterministic.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deIdentifyDeterministicEncryption de-identifies through deterministic encryption
+func deIdentifyDeterministicEncryption(w io.Writer, projectID, inputStr string, infoTypeNames []string, keyFileName, cryptoKeyName, surrogateInfoType string) error {
+	// projectId := "your-project-id"
+	// inputStr := "My SSN is 372819127"
```
Comment (telpirion) [inline dlp/snippets/deid/deid_deterministic.go:31]: question: Do we know that this is a dummy / fake SSN? If not, let's replace with something obviously fake, like `111111111`.

File: dlp/snippets/deid/deid_test.go [hunk 2]
```diff
@@ -263,3 +271,127 @@ func TestDeIdentifyWithWordList(t *testing.T) {
 		t.Errorf("deidentifyWithWordList(%q) = %q, want %q", input, got, want)
 	}
 }
+
+func TestDeIdentifyDeterministic(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	input := "Jack's phone number is 2255338899"
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:278]: question: Are we sure this is a fake phone number? Let's make sure to provide something that is obviously fake like `5555551212`.

File: dlp/snippets/deid/deid_test.go [hunk 3]
```diff
@@ -263,3 +271,127 @@ func TestDeIdentifyWithWordList(t *testing.T) {
...
+
+	if err := deIdentifyDeterministicEncryption(&buf, tc.ProjectID, input, infoTypeNames, keyFileName, cryptoKeyName, surrogateInfoType); err != nil {
+		t.Errorf("deIdentifyDeterministicEncryption(%q) = error '%q', want %q", err, input, want)
+	}
+
+	if got := buf.String(); !strings.Contains(got, want) {
+		t.Errorf("deIdentifyDeterministicEncryption(%q) = %q, want %q", input, got, want)
+	}
+
+	defer destroyKey(t, tc.ProjectID, keyVersion)
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:301]: issue(blocking): move this call to `defer` right to line 288. It's best practice to keep a destructor like this closer to what it's destroying.

**Review by annadata — APPROVED** [review]

File: dlp/snippets/deid/deid_test.go [hunk 4]
```diff
@@ -17,11 +17,18 @@ package deid
...
+	"fmt"
+	"log"
 	"strings"
 
 	"testing"
 
+	kms "cloud.google.com/go/kms/apiv1"
+	"cloud.google.com/go/kms/apiv1/kmspb"
 	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"github.com/gofrs/uuid"
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:31]: issue: use `github.com/google/uuid` instead.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid/deid_test.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: fixed it.

File: dlp/snippets/deid/deid_test.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: got your point, fixed it.

File: dlp/snippets/deid/deid_test.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: updated it.

File: dlp/snippets/deid/deid_deterministic.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_deterministic.go]: updated it.

**Pull request merged**